### ISSUE TITLE
mapper produces alignments with impossible coordinates

### ISSUE THREAD

**Issue by biork** [Illumina/DRAGMAP#12]
I am mapping some FASTQs using a built-from-source dragen-os on:

wget "ftp://hgdownload.cse.ucsc.edu/goldenPath/hg38/bigZips/p13/hg38.p13.fa.gz" with MD5
9930e3071308dc5e9b546934b43a2323  hg38.p13.fa.gz

I have discovered that an apparently problem-free run of the mapper generated lots of alignments (a count is in progress) with **start coordinates far beyond the target sequence's length**. 
Notably, this error only arises on the last sequence in hg38.p13.fa: chrX_ML143384v1_fix, suggesting some kind of search termination bug in dragen(?)

I am in the process of confirming this is the only affected sequence and will update, if it's not.

**Comment by Griffan** [comment]
Hi, @biork. I have tried to reproduce this problem using our internal dataset but failed. Could you please share with us your: version number, cmdline, and maybe a few fastq reads that are aligned to this contig? Thanks!

**Comment by biork** [comment]
Yes, I also tried to reproduce on semi-random data incorporating chrX_ML143384v1_fix and was  unable. I am now  regenerating exactly the data I used, and will update you when it's done.

**Comment by biork** [comment]
Hi Griffan,

It was easily reproduced once I regenerated the input FASTQs (extracted from earlier bwa alignments).

For what it's worth there is clearly a "theme" of repeat sequences with CCATT motifs in the incorrect output, though it's not present in all bad alignments.

dragen-os --version 1.2.1-2-ge4050868
using boost 1.69.0

LD_LIBRARY_PATH=../lib/boost/lib ../bin/dragen-os 
	-r ../resources/ucsc/hashtables 
	-1 sample.1.fq.gz 
	-2 sample.2.fq.gz 
	--num-threads 48 2> dragmap.err 
	| python3 ../bin/verify-mapping-coords.py 2> badalign.sam 
	| samtools view -b -o sample-nmord.bam &

Supplying reads is tricky since it's real data. I can probably safely provide a few, need to check, but not going to post those to github.

I'm good with gdb if there is debugging you'd like me to try on my end.
--
thanks,
roger kramer

**Comment by biork** [comment]
Ok, I have permission to hand off "a few" reads, but not via github.

**Comment by biork** [comment]
More information: running dragen with a new (different from above) reference and more real data, and again it is emitting impossible coordinates for _the last contig_ in the reference FASTA and _only the last contig_.
